content/token-swap.md

```diff
@@ -0,0 +1,848 @@
+# Token Swap
+
+# Lesson Objectives
+
+*By the end of this lesson, you will be able to:*
+
+- Create a token swap pool
+- Deposit liquidity
+- Withdraw liquidity
+- Swap tokens
+
+# TL;DR
+
+- The **token swap program** is an SPL contract deployed to Mainnet and Devnet available for use by developers and protocols.
+- The program accepts six different **instructions**, all of which we will explore in this lesson.
+- Developers are able to create and use **liquidity pools** to swap between any SPL token that they wish.
+- The program uses a mathematical formula called "**curve**" to calculate the price of all trades. Curves aim to mimic normal market dynamics: for example, as traders buy a lot of one token type, the value of the other token type goes up.
+
+# Overview
+
+## Swap Pools
+
+Before we get into how to create and interact with swap pools on Solana, it’s important we understand the basics of what a swap pool is. A swap pool is an aggregation of two different tokens with the purpose of providing liquidity to facilitate exchange between each token.
+
+Users provide liquidity to these pools by depositing their own tokens into each pool. These users are called liquidity providers. When a liquidity provider (or LP) deposits some tokens to the swap pool, LP-tokens are minted that represent the LP's fractional ownership in the pool. 
```

Comment:
@jamesrp13 there's a couple cleanup items - newlines and whitespaces.  I will add these to the PR